### PR TITLE
fix(pack): improve compatibility of table.unpack function

### PR DESCRIPTION
**Pull request by abhijeetsrivastava**
- found this fix from https://github.com/hrsh7th/nvim-cmp/issues/1017
- this fix is working fine in my local

### CODE REVIEW

File: lua/astrocommunity/pack/java/java.lua [hunk 1]
```diff
@@ -1,5 +1,8 @@
 local utils = require "astrocommunity.utils"
 
+-- 5.1 lua compatibility
+table.unpack = table.unpack or unpack
+
```
Comment (mehalter) [inline lua/astrocommunity/pack/java/java.lua:5]:
```suggestion
```

This causes Linting issues and we shouldn't override this function globally

**Review by mehalter — CHANGES_REQUESTED** [review]
We shouldn't override this function nlike this, we should change where it is called to just `(table.unpack or unpack)(...)`

File: lua/astrocommunity/pack/java/java.lua (hunk 1 above)
Comment (abhijeetsrivastava) [reply lua/astrocommunity/pack/java/java.lua:5]: makes sense. I will update

**Review by mehalter — APPROVED** [review]
Great! Thanks for this!

**Pull request merged**